include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -166,9 +273,119 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
     }
 };
 
+template <typename... _Name>
+class _find_split_points_kernel_on_mid_diagonal_uint32_t;
+
+template <typename... _Name>
+class _find_split_points_kernel_on_mid_diagonal_uint64_t;
+
+template <typename _IdType, typename _CustomName, typename... _Name>
+struct __parallel_merge_submitter_large<_IdType, _CustomName, __internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Compare>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp) const
+    {
+        const _IdType __n1 = __rng1.size();
+        const _IdType __n2 = __rng2.size();
+        const _IdType __n = __n1 + __n2;
+
+        assert(__n1 > 0 || __n2 > 0);
+
+        _PRINT_INFO_IN_DEBUG_MODE(__exec);
+
+        using _FindSplitPointsOnMidDiagonalKernel =
```

Comment:
@rarutyun I have fixed the error here. Is it correct way?
I am using `__kernel_name_generator` here because I should have two Kernel names: one passed as template parameter pack and the second name I should create inside.

Reply:
Will discuss with @rarutyun but as far as I understood it's not possible because we should have mandatory type pack as template param for the class like
```C++
template <typename... _Name>
class _find_split_points_kernel_on_mid_diagonal;
```

Reply:
Aha, this is actually an interesting one because the type differs based on a runtime parameter rather than the types the user provides, and so we need both for a single public API call in the code.  I see why its necessary and we cant just add it to the pack now. 

```suggestion
        // Runtime decision to pick the _IdType, so we need both kernels for a single public call, and must embed that in the base name
        using _KernelNameWithType = std::conditional_t<std::is_same_v<_IdType, std::uint32_t>, _find_split_points_kernel_on_mid_diagonal_uint32_t, _find_split_points_kernel_on_mid_diagonal_uint64_t>;
        using _FindSplitPointsOnMidDiagonalKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<_KernelNameWithType,_CustomName, _ExecutionPolicy,
                                   _Range1, _Range2, _Range3, _Compare>;
```

Reply:
Unfortunately this approach has compile errors: discussed offline.